src/Matcher.js

```diff
@@ -0,0 +1,72 @@
+const { getData } = require('./Fetcher');
+const { writeJson, readJson } = require('./Json');
+const { crawlDataSkin } = require('./Crawler');
+const { credentials } = require('../config.json');
+const log = require('node-pretty-log');
+
+
+const getUsedWidgets = slots => {
+    let widgets = [];
+    Object.values(slots).forEach(slot => {
+        widgets = [...widgets, ...slot.widgets || []];
+    });
+    return widgets;
+};
+
+const crawlThroughInfo = (dataSkins, centralInfo) => {
+    centralInfo.forEach(centralConfig => {
+        const widgets = getUsedWidgets(centralConfig.slots);
+        widgets.forEach(widget => {
+            const widgetObject = centralConfig.widgets[widget];
+            const specifiedWidget = dataSkins[widgetObject.feature];
+            const widgetSkin = widgetObject.skin || 'default';
+            const isWidgetEnabled = widgetObject.enabled;
+            if (specifiedWidget[widgetSkin]) {
+                specifiedWidget[widgetSkin]
+                    .enabled = specifiedWidget[widgetSkin].enabled || isWidgetEnabled;
+            } else {
+                specifiedWidget.notListed = specifiedWidget.notListed || {};
+                specifiedWidget.notListed[widgetSkin] = [
+                    ...(specifiedWidget.notListed[widgetSkin] || []),
+                    { location: centralConfig.config.name, enabled: isWidgetEnabled },
+                ];
+            }
+        });
+    });
+    return dataSkins;
```

Comment:
Here you are modifying `centralInfo` and just using `dataSkins` then returning it, is it right?

Reply:
I'm modifying through specifiedWidget